haicrypt/hcrypt_tx.c

```diff
@@ -34,11 +34,13 @@ written by
 int HaiCrypt_Tx_GetBuf(HaiCrypt_Handle hhc, size_t data_len, unsigned char **in_pp)
 {
 	hcrypt_Session *crypto = (hcrypt_Session *)hhc;
-	int pad_factor = (HCRYPT_CTX_MODE_AESECB == crypto->ctx->mode ? 128/8 : 1);
+	int pad_factor;
 
 	ASSERT(NULL != crypto);
 	ASSERT(NULL != crypto->cipher);
 
+        pad_factor = (HCRYPT_CTX_MODE_AESECB == crypto->ctx->mode ? 128/8 : 1);
```

Comment:
No need to change the original line
```
int pad_factor = (HCRYPT_CTX_MODE_AESECB == crypto->ctx->mode ? 128/8 : 1);
```
Just move it below the assertions.

Reply:
okay.